rtm/plotting.py

```diff
@@ -151,8 +151,9 @@ def plot_time_slice(S, processed_st, time_slice=None, label_stations=True,
         plot_transform = ax.transData
 
         # Mask areas outside of DEM extent
-        slice.data[np.isnan(dem.data)] = np.nan
-
+        dem_slice = dem.sel(x=slice.x, y=slice.y, method='nearest')
```

Comment:
Just adding a comment, change if I'm not interpreting this right
```suggestion
        dem_slice = dem.sel(x=slice.x, y=slice.y, method='nearest')  # Select subset of DEM that slice occupies
```